export_job/export_zip_from_manifest_job.py

```diff
@@ -0,0 +1,184 @@
+)
+
+
+async def build_manifest_from_study_ids(hostname, token, study_ids):
+    """
+    build a manifest from a list of metadata guids representing study ids
+    dumps result to gen3-sdk compatible manifest file
+    """
+    ongoing_requests = []
+    study_metadata = []
+
+    print(f"Assembling manifest for study ids: {study_ids}")
+
+    async with aiohttp.ClientSession() as client:
+        for study_id in study_ids:
+            ongoing_requests += [
+                client.get(
+                    f"https://{hostname}/mds/aggregate/metadata/guid/{study_id}",
+                    headers={"Authorization": f"bearer {token}"},
+                )
+            ]
+            if len(ongoing_requests) == CHUNK_SIZE:
+                study_metadata += await asyncio.gather(
+                    *[req.json() for req in await asyncio.gather(*ongoing_requests)]
+                )
+
+        study_metadata += await asyncio.gather(
+            *[req.json() for req in await asyncio.gather(*ongoing_requests)]
+        )
+
+    # print(f"Retrieved metadata: {study_metadata}")
+    manifest = []
+    for study in study_metadata:
+        if not study["__manifest"]:
+            print(
+                f"Study {study['project_number']} is missing __manifest entry. Skipping."
+            )
+        else:
+            manifest += study["__manifest"]
+
+    download_size = sum(file.get("file_size", 0) for file in manifest)
+    if download_size > MAX_DOWNLOAD_SIZE:
+        fail(
+            f"The selected studies contain {download_size / 1000000} MB of data, "
+            "which exceeds the download limit of 250 MB. "
+            "Please deselect some studies and try again, or use the Gen3 client."
+        )
+
+    with open(MANIFEST_FILENAME, "w+") as f:
+        json.dump(manifest, f)
+
+
+def download_files(access_token, hostname):
+    """
+    download the files described by the local manifest and export them to a zip
+    """
+    with TemporaryAPIKey(token=access_token, hostname=hostname):
+        auth = Gen3Auth(refresh_file=TemporaryAPIKey.file_name)
+        list_files_in_workspace_manifest(MANIFEST_FILENAME)
+        describe_access_to_files_in_workspace_manifest(
+            hostname, auth, MANIFEST_FILENAME
+        )
+        download_files_in_workspace_manifest(
+            hostname, auth, MANIFEST_FILENAME, EXPORT_DIR
+        )
+        shutil.make_archive(EXPORT_DIR, "zip", EXPORT_DIR)
+
+
+def upload_export_to_s3(s3_credentials, username):
+    """
+    uploads the local zip export to s3 and returns a presigned url, expires after 1 hour
+    """
+    bucket_name = s3_credentials["bucket_name"]
+    aws_access_key_id = s3_credentials["aws_access_key_id"]
+    aws_secret_access_key = s3_credentials["aws_secret_access_key"]
+
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+
+    export_key = f"{username}-export.zip"
```

Comment:
Is there a risk/possibility that an IDP may allow a user to be authenticated, but no `username` is set? Or could the username have special characters that don't play well with S3 or various file systems?

Reply:
These are both good questions, but I'm not sure what fields an IDP requires or what characters they can take on. Can you think of a more appropriate way to name exports? I'm brainstorming...

Reply:
A `guid` might be a nice way to simply namespace files. That also allows us to not have to worry about storing names/identifying information in S3 or anywhere else

A timestamp + guid is also nice because it allows us to see how old exports are and expire them while enforcing uniqueness